app/app.go

```diff
@@ -45,29 +75,69 @@ func NewApp(cfg *config.Config) *App {
 
 	hashVerifier := verifier.NewHashVerifier(cfg, daoManager, executor, cfg.GreenfieldConfig.DeduplicationInterval)
 
-	signer := vote.NewVoteSigner(ethcommon.Hex2Bytes(cfg.VotePoolConfig.BlsPrivateKey))
+	signer := vote.NewVoteSigner(executor.BlsPrivKey)
 	voteDataHandler := vote.NewDataHandler(daoManager, executor)
-	voteProcessor := vote.NewVoteProcessor(cfg, daoManager, signer, executor, voteDataHandler)
+	voteCollector := vote.NewVoteCollector(cfg, daoManager, executor, voteDataHandler)
+	voteBroadcaster := vote.NewVoteBroadcaster(cfg, daoManager, signer, executor, voteDataHandler)
+	voteCollator := vote.NewVoteCollator(cfg, daoManager, signer, executor, voteDataHandler)
 
 	txDataHandler := submitter.NewDataHandler(daoManager, executor)
 	txSubmitter := submitter.NewTxSubmitter(cfg, executor, daoManager, txDataHandler)
 
+	dbWiper := wiper.NewDBWiper(daoManager)
+
 	return &App{
-		executor:      executor,
-		eventMonitor:  monitor,
-		hashVerifier:  hashVerifier,
-		voteProcessor: voteProcessor,
-		txSubmitter:   txSubmitter,
+		executor:        executor,
+		eventMonitor:    monitor,
+		hashVerifier:    hashVerifier,
+		voteCollector:   voteCollector,
+		voteBroadcaster: voteBroadcaster,
+		voteCollator:    voteCollator,
+		txSubmitter:     txSubmitter,
+		dbWiper:         dbWiper,
```

Comment:
Do we need wiper? Sometimes we need to review or query historical data. We may not need it now.